UI: Show Zulip Version in the app

Signior-X opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Follow-up PR for #17254 
This PR needs some suggestions and reviews of how the behaviour is intended.
[CZO conversation here.](https://chat.zulip.org/#narrow/stream/2-general/topic/Issue.20.238005.3A.20Make.20version.20number.20visible.20in.20UI/near/1134800)

Fixes #8005



**Testing plan:** <!-- How have you tested? -->
Tested manually currently.

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
The gear menu.
![image](https://user-images.githubusercontent.com/56730716/111552474-6b563780-87a8-11eb-8acb-294513b78ca2.png)

The overlay showing information. The first line shows the zulip_version and the second line is expected to be zulip_version_name or something.
![image](https://user-images.githubusercontent.com/56730716/111552556-99d41280-87a8-11eb-8492-c08e884d5e12.png)



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented on templates/zerver/app/navbar.html:
```diff
@@ -104,6 +104,12 @@
+                            <li role="presentation">
+                                <a href="https://zulip.readthedocs.io/en/latest/overview/changelog.html"  target="_blank" rel="noopener noreff" role="menuitem" class="menuitem-version">
+                                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="49.99 49.99 673.14 673.14"><path d="M688.52 150.67c0 33.91-15.23 64.04-38.44 82.31L424.79 434.17c-4.18 3.59-9.62-2.19-6.61-7.03l82.64-165.46c2.31-4.63-.69-10.33-5.44-10.33H174.86c-49.64 0-90.26-45.31-90.26-100.68 0-55.37 40.62-100.68 90.26-100.68h423.39c49.65 0 90.27 45.31 90.27 100.68zM174.86 723.13h423.39c49.64 ...
+                                    {{ _('Zulip ' + (zulip_version[:-4] if zulip_version.endswith('+git') else zulip_version) ) }}
```
See the comments I've posted in the original PR -- I want a function in say `zerver/lib/home.py` that generates a URL as well as a string to use that will be passed into this template -- we'll end up having like 5 conditionals in that function, I think.

Signior-X replied:
Yes that's why I created the variable `zulip_version_name`. I need to know just this. Working on this, will update soon.

Signior-X replied:
Hello, I have created the function in `context_processors.py`, I have currently made a simple logic for the name to be shown. I don't know the conditionals, either you can add them accordingly or tell about them.
Also what URL are you talking about to be generated in the function? I didn't get that.

Signior-X commented:
@timabbott I have updated the PR, added the function to generate `zulip_verision_name`.

Tasks left according to me
- [x] Make the design of the overlay better by taking feedback from people in CZO.
- [ ] Add more conditions to the `get_zulip_version_name` function.
- [ ] After adding the conditions, write some tests for that function.

Do tell me what to do next, and what URL you are talking about.

Signior-X commented:
I have rebased the current PR as conflicts arose. This PR is ready for review @timabbott

Signior-X commented:
@timabbott can you suggest what to do now in this?

timabbott commented:
@Signior-X OK I think my proposal is to split this PR a bit -- first, let's make a backend function to return the two values (and write a test suite for it) -- you can do this in a new PR for simplicity, and we'll work to merge that (I think it'll take some rounds of review).  And then we can come back to the display UI for it -- I think your prototype is pretty good but all the hard thinking will be in my thinking through the ~4 cases for the backend function, which I can do pretty quickly if it's trivial to add a test for a new case.  Does that make sense?

timabbott commented:
I'd like to take a new strategy with this -- let's get something simple (without all the cases) merged, enabled only in a development environment.  And then multiple folks can do iterative improvements on this over the next few days.

I've pushed back here after a half-dozen cleanups:
* Removed links I'm not sure we want
* Changed the menu item to only be visible in development

The things I'd like to do before merging an initial version:
* Remove the inline Zulip logo SVG; we should have an external asset we can use for that.  
* Don't use the favicon image for the logo inside the version notice; it's low-resolution.  We should be able to just use our normal SVG.  
* We should get this passing CI.

And then it'll be possible for me to iterate on the content of the popup with something that actually runs.

Signior-X commented:
Sure, working on this. Will update soon.

amanagr commented:
@Signior-X I have started working on finishing this.

Signior-X commented:
oh @amanagr sure, no worries.

amanagr commented:
<img width="545" alt="Screenshot 2021-05-01 at 4 22 35 AM" src="https://user-images.githubusercontent.com/25124304/116762122-16f4d780-aa37-11eb-9b30-64fa6551c0fa.png">

Do you want me to use the clipboard icon we use elsewhere? I used it because I like this one better and goes with a more minimalistic design.

timabbott commented:
I merged this after a few tweaks:
* Links with `target="_blank"` must have `rel=nopener` for security.
* Small wording changes.

I left #8005 open. since this isn't enabled in production yet, but the work on this is appreciated!

The main remaining work here is adding more content to that widget, and also making sure we like the version math.

Thanks folks!

kfogel commented:
Saw that issue #8005 is now closed, came here, looked at the new widget content, and was glad!  Thank you, @Signior-X, @amanagr, and @timabbott.

Signior-X commented:
@kfogel special thanks to @pilgrim2308 who initiated this work and created a very good layout on how to do this.

kfogel commented:
Credit where credit is due!  Good of you to point that out; thank you.
